ext/intl/calendar/calendar_methods.cpp

```diff
@@ -388,38 +389,38 @@ U_CFUNC PHP_FUNCTION(intlcal_set)
 	int arg_num = ZEND_NUM_ARGS() - (object ? 0 : 1);
 
 	if (zend_parse_method_parameters(
-		ZEND_NUM_ARGS(), object, "Oll|llll",
-		&object, Calendar_ce_ptr, &args[0], &args[1], &args[2], &args[3], &args[4], &args[5]) == FAILURE
-	) {
+		ZEND_NUM_ARGS(), object, "Oll|l!l!l!l!",
+		&object, Calendar_ce_ptr, &args[0], &args[1], &args[2], &args_is_null[2],
+		&args[3], &args_is_null[3], &args[4], &args_is_null[4], &args[5], &args_is_null[5]
+	) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	for (int i = 0; i < arg_num; i++) {
-		if (args[i] < INT32_MIN || args[i] > INT32_MAX) {
+		if ((i < 2 || !args_is_null[i]) && (args[i] < INT32_MIN || args[i] > INT32_MAX)) {
 			intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR,
-				"intlcal_set: at least one of the arguments has an absolute "
-				"value that is too large", 0);
+				"intlcal_set: at least one of the arguments has an absolute value that is too large", 0);
 			RETURN_FALSE;
 		}
 	}
 
 	CALENDAR_METHOD_FETCH_OBJECT;
 
-	if (arg_num == 2) {
+	if (!args_is_null[5]) {
```

Comment:
This code doesn't look right to me. If the last one is not null, but some before it are, isn't this going to use an uninitialized value?

Reply:
I believe this issue is still unresolved. And I'm not sure what can be done here, given how the API looks like. I'd probably just keep the argument count checks for this case...

Reply:
Now I understood what you meant here (first, I though that you still refer to the uninitialized `args_is_null` array).

Keeping the argument number checks also means that we shouldn't add default values, right? So then it's better if I revert this change. :(

Reply:
Another idea came to my mind: We could define proper default values if we also checked whether there are only "trailing" `null`s given (so there are no `null`s before any non-null value). If this is not the case, a similar exception to the `No variant with 4 arguments` could be thrown. What do you think? Is it ok to do so?

Reply:
That would be okay, but I'm not sure it buys us anything. You still wouldn't be able to pass the default value in any meaningful way.

Reply:
I agree that this is far from perfect :)